src/modules/roc_core/target_posix/roc_core/format_time.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <stdio.h>
+#include <sys/time.h>
+#include <time.h>
+
+#include "roc_core/format_time.h"
+
+namespace roc {
+namespace core {
+
+bool format_time(char* buf, size_t bufsz) {
+    timeval tv;
+    if (gettimeofday(&tv, NULL) == -1) {
+        return false;
+    }
+
+    tm t;
+    if (!localtime_r(&tv.tv_sec, &t)) {
+        return false;
+    }
+
+    size_t off = strftime(buf, bufsz, "%H:%M:%S", &t);
+    if (off == 0) {
+        return false;
+    }
+
+    if (off == bufsz) {
```

Comment:
Is it possible?
If we provide too small buffer `strftime` returns 0, isn't it?

Reply:
Does it handle the case when bufsz == 0?

Reply:
It's not possible. And even if it would, it's not needed anyway because snprintf() handles the case where bufsz is zero. So I removed the check.

Reply:
Hm, no, snprintf() handles zero size specially. It would be incorrect to pass zero in our case. Though according to strftime() spec this case is not possible, I returned the check, just in case.